wallets/exodus-extension/src/extension/main-wallet.ts

```diff
@@ -0,0 +1,24 @@
+import { EndpointOptions, Wallet } from '@cosmos-kit/core';
+import { MainWalletBase } from '@cosmos-kit/core';
+
+import { ChainExodusExtension } from './chain-wallet';
+import { ExodusClient } from './client';
+import { getExodusFromExtension } from './utils';
+
+export class ExodusExtensionWallet extends MainWalletBase {
+  constructor(walletInfo: Wallet, preferredEndpoints?: EndpointOptions) {
+    super(walletInfo, ChainExodusExtension);
+    this.preferredEndpoints = preferredEndpoints;
+  }
+
+  async initClient() {
+    this.initingClient();
+    try {
+      const exodus = await getExodusFromExtension();
+      this.initClientDone(exodus ? new ExodusClient(exodus.cosmos) : undefined);
```

Comment:
I'm wondering if we should instead do `exodus?.cosmos` -- it __seems__ safer to me in case someone has a version of the extension w/o the cosmos provider.   not sure if that will really happen though as ideally, we always have the cosmos object injected into our namespace.